std/traits.d

```diff
@@ -5364,16 +5369,23 @@ enum bool isNumeric(T) = is(NumericTypeOf!T) && !isAggregateType!T;
 /**
 Detect whether $(D T) is a scalar type (a built-in numeric, character or boolean type).
  */
-enum bool isScalarType(T) = isNumeric!T || isSomeChar!T || isBoolean!T;
+enum bool isScalarType(T) = is(T : real);
```

Comment:
With that in mind, there should probably be some tests with structs with `alias this` to make sure that they don't get treated as what they convert to by these traits.